### PR TITLE
feat(ci): add slack notification when betas fail

### PR DESCRIPTION
**Pull request by brainbicycle**
The type of this PR is: **Enhancement**

### Description

Adds a slack notification when beta lane fails for any reason based on discussion in retro and feedback at front-end-ios (now mobile) practice.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -211,6 +211,22 @@ lane :notify_if_new_license_agreement do
   end
 end
 
+lane :notify_beta_failed do |options|
+  exception = options[:exception]
+  message = ':x: :iphone:\n'\
+            'Looks like the latest beta failed to deploy!\n'\
+            'See circle job for more details.'
```
Comment (ashfurrow) [inline fastlane/Fastfile:218]: Ruby 2.3 added support for HEREDOC-style multiline strings that unindent for you. The equivalent code using them would be:

```suggestion
  message = <<~MSG
    :x: :iphone:
    Looks like the latest beta failed to deploy!
    See circle job for more details.
  MSG
```

Older Ruby versions didn't have `<<~`, only `<<-`, which would have added extra spaces at the beginning of each line.

File: fastlane/Fastfile [hunk 2]
```diff
@@ -211,6 +211,22 @@ lane :notify_if_new_license_agreement do
...
+  slack(
+    message: message,
+    success: false,
+    payload: {
+      'Circle Build' => ENV['CIRCLE_BUILD_URL'],
```
Comment (ashfurrow) [inline fastlane/Fastfile:223]: Yes! Good idea.

**Review by ashfurrow — APPROVED** [review]
Great! Just one suggestions about Ruby syntax 👍

**Review by sepans — APPROVED** [review]
Thanks for adding this so quickly 🙏 😍

File: fastlane/Fastfile (hunk 1 above)
Comment (brainbicycle) [reply fastlane/Fastfile:218]: ✨ Nice! I'll add that to the other slack message as well

**Pull request merged**